File: fluttermultiplatform/lib/models/review.dart
import 'user.dart';

class Review {
  final String id;
  final String userId;
  final String productId;
  final int rating;
  final String comment;
  final User? user;
  final String createdAt;

  Review({required this.id, this.userId = '', this.productId = '', this.rating = 0, this.comment = '', this.user, this.createdAt = ''});

  factory Review.fromJson(Map<String, dynamic> json) => Review(
        id: json['id']?.toString() ?? '',
        userId: json['user_id']?.toString() ?? '',
        productId: json['product_id']?.toString() ?? '',
        rating: (json['rating'] ?? 0).toInt(),
        comment: json['comment'] ?? '',
        user: json['user'] != null ? User.fromJson(json['user']) : null,
        createdAt: json['created_at'] ?? '',
      );
}

File: backend/models/review.go
package models
import (
	"time"
	"github.com/google/uuid"
	"gorm.io/gorm"
)
type Review struct {
	ID        uuid.UUID      `gorm:"type:uuid;primary_key" json:"id"`
	UserID    uuid.UUID      `gorm:"type:uuid;not null;index" json:"user_id"`
	ProductID uuid.UUID      `gorm:"type:uuid;not null;index" json:"product_id"`
	Rating    int            `gorm:"not null" json:"rating" validate:"required,min=1,max=5"`
	Comment   string         `gorm:"type:text" json:"comment"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
	User    User    `gorm:"foreignKey:UserID" json:"user,omitempty"`
	Product Product `gorm:"foreignKey:ProductID" json:"product,omitempty"`
}
func (r *Review) BeforeCreate(tx *gorm.DB) error {
	if r.ID == uuid.Nil {
		r.ID = uuid.New()
	}
	return nil
}
